### PR TITLE
LITE-16977 Create Usage Functionality

### PR DESCRIPTION
**Pull request by martinconstante**

### CODE REVIEW

File: lib/connect/api/assets.js [hunk 1]
```diff
@@ -4,8 +4,20 @@
...
 const GenericResource = require('./base');
 
+/**
+ * The *AssetUsageAgregatesResource* class provides methods to access the
+ * *AssetUsageAgregates* objects for a asset.
+ *
+ * @extends GenericResource
+ * @category Resources
+ */
+class AssetUsageAgregates extends GenericResource {}
```
Comment (ffaraoneim) [inline lib/connect/api/assets.js:19]: Shoud be AssetUsageAggregateResource

File: lib/connect/api/products.js [hunk 2]
```diff
@@ -97,6 +97,15 @@ class ProductParameterResource extends GenericResource {}
...
 class ProductConnectionsResource extends GenericResource {}
 
+/**
+ * The *ProductUsageRulesResource* class provides methods to access the
+ * *ProductUsageRule* objects for a product.
+ *
+ * @extends GenericResource
+ * @category Resources
+ */
+class ProductUsageRulesResource extends GenericResource {}
```
Comment (ffaraoneim) [inline lib/connect/api/products.js:107]: Better ProductUsageRuleResource

File: lib/connect/api/usageReconciliations.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+const GenericResource = require('./base');
+
+/**
+ * The *UsageFileResource* class provides methods to access the usage file
+ * endpoint of the Cloud Blue Connect API.
+ *
+ * @extends GenericResource
+ * @category Resources
+ */
+class UsageReconciliationsResource extends GenericResource {
```
Comment (ffaraoneim) [inline lib/connect/api/usageReconciliations.js:18]: Better UsageReconciliationResource

File: lib/connect/api/usageReconciliations.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  /**
+   * Get processed files.
+   *
+   * @param   {string}  id                  The unique identifier of the *Usage File*.
+   * @returns {UsageReconciliationResource} An instance of the *UsageReconciliationResource*.
+   *
+   */
+  async processedfile(id) {
```
Comment (ffaraoneim) [inline lib/connect/api/usageReconciliations.js:37]: Better processedFile

File: lib/connect/api/usageReconciliations.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  /**
+   * Get uploaded files.
+   *
+   * @param   {string}  id                  The unique identifier of the *Usage File*.
+   * @returns {UsageReconciliationResource} An instance of the *UsageReconciliationResource*.
+   *
+   */
+  async uploadedfile(id) {
```
Comment (ffaraoneim) [inline lib/connect/api/usageReconciliations.js:53]: Better uploadedFile

**Review by ffaraoneim — CHANGES_REQUESTED** [review]

File: lib/connect/api/assets.js (hunk 1 above)
Comment (martinconstante) [reply lib/connect/api/assets.js:19]: Done

File: lib/connect/api/products.js (hunk 2 above)
Comment (martinconstante) [reply lib/connect/api/products.js:107]: Done

File: lib/connect/api/usageReconciliations.js (hunk 3 above)
Comment (martinconstante) [reply lib/connect/api/usageReconciliations.js:18]: Done

File: lib/connect/api/usageReconciliations.js (hunk 4 above)
Comment (martinconstante) [reply lib/connect/api/usageReconciliations.js:37]: Done

File: lib/connect/api/usageReconciliations.js (hunk 5 above)
Comment (martinconstante) [reply lib/connect/api/usageReconciliations.js:53]: Done

**Review by ffaraoneim — APPROVED** [review]

**Pull request merged**